pkg/controller/backupstoragelocation_controller.go

```diff
@@ -132,12 +152,15 @@ func (r *BackupStorageLocationReconciler) logReconciledPhase(defaultFound bool,
 	}
 
 	if !defaultFound {
-		log.Warnf("The specified default backup location named %q was not found; for convenience, be sure to create one or make another backup location that is available the default", r.StorageLocation.DefaultStorageLocation)
+		log.Warnf("The specified default backup location named %q was not found; for convenience, be sure to create one or make another backup location that is available the default", r.DefaultBackupLocationInfo.DefaultStorageLocation)
 	}
 }
 
 func (r *BackupStorageLocationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&velerov1api.BackupStorageLocation{}).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: 10,
```

Comment:
The BSL controller, in every execution of the `Reconcile` method is validating all BSLs in the cluster based on their validation frequency. So running multiple reconciles at the same time will imply that we are concurrently validating all BSL which might result in unnecessary API calls to the BSL and also to the API server, from the `patch` calls.

To prevent delays in the reconciliation of BSLs, we should be running iterations of the loop on Line 65 in parallel and not run multiple reconciles in parallel.

Reply:
This is a good point. As written, this call means [the loop](https://github.com/vmware-tanzu/velero/pull/2870/files#diff-cfcbc9b9778d96e266dd99924a91c426R65) is effectively being called 10 different times in parallel, iterating over each BSL 10 times at once.

So to fan-out, we'd change to take L65-L89 and make most of it the contents of a go routine that would receive an individual BSL, then have some number of concurrent goroutines to process the BSLs.

However, I think that would be an optimization and is not a requirement for shipping this. _Most_ installs aren't going to have tons of BSLs, but this would be a good pattern support in our controllers that do have lots of objects, like backups and restores.

Reply:
I removed the `MaxConcurrentReconciles` and created a new issue: https://github.com/vmware-tanzu/velero/issues/2883